File: librored/backend/src/main/java/es/codeurjc13/librored/dto/LoanDTO.java
package es.codeurjc13.librored.dto;

import es.codeurjc13.librored.model.Loan;
import java.time.LocalDate;

/**
 * Loan DTO for REST API operations
 * Contains complete loan information with basic references to related entities
 */
public record LoanDTO(
        Long id,
        BookBasicDTO book,
        UserBasicDTO lender,
        UserBasicDTO borrower,
        LocalDate startDate,
        LocalDate endDate,
        Loan.Status status) {
}

File: librored/frontend/src/app/dtos/loan.dto.ts
export interface BookBasicDTO {
  id: number;
  title: string;
  author: string;
}

export interface UserBasicDTO {
  id: number;
  username: string;
}

export interface LoanDTO {
  id?: number;
  book: BookBasicDTO;
  lender: UserBasicDTO;
  borrower: UserBasicDTO;
  startDate: string; // Format: YYYY-MM-DD
  endDate?: string; // Format: YYYY-MM-DD, optional
  status: LoanStatus;
}

export enum LoanStatus {
  Active = 'Active',
  Completed = 'Completed'
}

export interface LoanRequest {
  book: BookBasicDTO;
  lender: UserBasicDTO;
  borrower: UserBasicDTO;
  startDate: string;
  endDate?: string;
  status: LoanStatus;
}